pyanaconda/isignal.py

```diff
@@ -0,0 +1,54 @@
+""" A signal/slot implementation
+
+File:    isignal.py
+Author:  Thiago Marcos P. Santos
+Author:  Christopher S. Case
+Author:  David H. Bronke
+Created: August 28, 2008
+Updated: December 12, 2011
+License: MIT
+
+"""
+import inspect
+from weakref import WeakKeyDictionary
+
+
+class Signal(object):
+    def __init__(self):
+        # The original implementation used WeakSet to store functions,
+        # but that causes lambdas without any other reference to be
+        # garbage collected. So we use a normal set to avoid that.
+        self._functions = set()
+        self._methods = WeakKeyDictionary()
+
+    def __call__(self, *args, **kargs):
```

Comment:
While I still like how you can just call the signal itself, I do agree that a separate `emit()` methods is more consistent. We already have `connect()`, `disconnect()` and `clear()`, so also having a method for triggering the signal makes sense.

Looks like [even GTK3 has an `emit()` method](http://python-gtk-3-tutorial.readthedocs.io/en/latest/objects.html#create-new-signals) for signals used from Python. :)